financial-services-accelerator/components/org.wso2.financial.services.accelerator.common/src/main/java/org/wso2/financial/services/accelerator/common/util/HTTPClientUtils.java

```diff
@@ -59,85 +59,110 @@ public class HTTPClientUtils {
     public static final String HTTPS_PROTOCOL = "https";
     private static final String[] SUPPORTED_HTTP_PROTOCOLS = {"TLSv1.2"};
     private static final Log log = LogFactory.getLog(DatabaseUtils.class);
+    private static PoolingHttpClientConnectionManager connectionManager;
+
+    private HTTPClientUtils() {
+        // Prevent instantiation
+    }
 
     /**
-     * Get closeable https client.
+     * Initialize the connection manager for HTTPS protocol.
      *
-     * @return Closeable https client
+     * @param maxTotal     Maximum pool connections
+     * @param maxPerRoute Maximum connections per route
      * @throws FinancialServicesException FinancialServicesException exception
      */
-    @Generated(message = "Ignoring since method contains no logics")
-    public static CloseableHttpClient getHttpsClient() throws FinancialServicesException {
+    private static void initConnectionManagerForHttpsProtocol(int maxTotal, int maxPerRoute)
+            throws FinancialServicesException {
 
         SSLConnectionSocketFactory sslsf = createSSLConnectionSocketFactory();
 
-        Registry<ConnectionSocketFactory> socketFactoryRegistry = RegistryBuilder.<ConnectionSocketFactory>create()
+        Registry<ConnectionSocketFactory> socketFactoryRegistry = RegistryBuilder
+                .<ConnectionSocketFactory>create()
                 .register(HTTP_PROTOCOL, new PlainConnectionSocketFactory())
                 .register(HTTPS_PROTOCOL, sslsf)
                 .build();
 
-        final PoolingHttpClientConnectionManager connectionManager = (socketFactoryRegistry != null) ?
-                new PoolingHttpClientConnectionManager(socketFactoryRegistry) :
-                new PoolingHttpClientConnectionManager();
-
-        // configuring default maximum connections
-        connectionManager.setMaxTotal(FinancialServicesConfigParser.getInstance().getConnectionPoolMaxConnections());
-        connectionManager.setDefaultMaxPerRoute(FinancialServicesConfigParser.getInstance()
-                .getConnectionPoolMaxConnectionsPerRoute());
+        connectionManager = new PoolingHttpClientConnectionManager(socketFactoryRegistry);
```

Comment:
Shall we add lazy holder pattern for connectionManager also,so it's refer from HTTPClientHolder